enhancements/machine-api/control-plane-set.md

```diff
@@ -0,0 +1,603 @@
+Machine, wait for this to become ready, and then remove the old Machine. It will repeat this process until all Machines
+are up to date.
+
+At first, we will not allow any configuration of the RollingUpdate (as you might see in a Deployment) and will pin the
+surge to 1 replica. We may wish to change this later once we have more data about the stability of this operator and
+the etcd protection mechanism it relies on.
+
+We expect this strategy to be used in most applications of the ControlPlaneSet, in particular it can only not be used
+in environments where capacity is very limited and a surge of any control plane capacity is unavailable.
+
+##### The Recreate update strategy
+
+The Recreate strategy mirrors the Recreate strategy familiar to users from Deployments.
+When a change is required to be rolled out, the Machine controller will first remove a Control Plane Machine, wait for
+its removal, and then create a new Control Plane Machine.
+
+This strategy is intended to be used only in very resource constrained environments where there is no capacity to
+introduce an extra temporary Control Plane Machine (preventing the RollingUpdate strategy).
+
+At present, when using this strategy, the updates will need manual intervention. The etcd protection design will
+prevent the Control Plane Machine from being removed until a replacement for the etcd member has been introduced into
+the cluster. To allow for this use case, the end user can manually remove the etcd protection (via removing the Machine
+deletion hook on the deleted Machine) allowing the rollout to proceed, the etcd operator will not re-add the
+protection if the Machine is already marked for deletion.
+
+This strategy introduces risk into the update of the Machines. While the old Machine is being drained/removed and the
+new Machine is being provisioned, etcd quorum is at risk as a member has been removed from the cluster. In most
+circumstances this would leave the cluster with just 2 etcd members.
+
+There are a number of open questions related to this update strategy:
+- Are we comfortable offering this strategy given the risks that it poses?
+- What alternative can we provide to satisfy resource constrained environments if we do not implement this strategy?
+- Should we teach the etcd operator to remove the protection mechanism when the ControlPlaneSet is configured with this
+  strategy?
+- To minimise risk, should we allow this strategy only on certain platforms? (Eg disallow the strategy on public clouds)
+
+##### The OnDelete update strategy
+
+The OnDelete strategy mirrors the OnDelete strategy familiar to users from StatefulSets.
+When a change is required, any logic for rolling out the Machine will be paused until the ControlPlaneSet controller
+observes a deleted Machine.
+When a Machine is marked for deletion, it will create a new Machine based on the current spec.
+It will then proceed with the replacement as normal.
```

Comment:
A `Machine` should never be in the cluster if it isn't gone from the provider, so that's how we would know.

For resource constrained environments we have the Recreate strategy, this strategy is just for trialing new configs or slowing down the rollout by manually choosing when each control plane member is replaced